packages/subproviders/src/subproviders/nonce_tracker.ts

```diff
@@ -0,0 +1,107 @@
+import * as _ from 'lodash';
+
+import EthereumTx = require('ethereumjs-tx');
+import ethUtil = require('ethereumjs-util');
+import providerEngineUtils = require('web3-provider-engine/util/rpc-cache-utils');
+
+import { JSONRPCPayload } from '../types';
+
+import { Subprovider } from './subprovider';
+
+const NONCE_TOO_LOW_ERROR_MESSAGE = 'Transaction nonce is too low';
+
+export class NonceTrackerSubprovider extends Subprovider {
+    private _nonceCache: { [address: string]: string } = {};
+    private static _reconstructTransaction(payload: JSONRPCPayload): EthereumTx {
+        const raw = payload.params[0];
+        const transactionData = ethUtil.stripHexPrefix(raw);
```

Comment:
I think we can reduce these two lines to one `ethUtil.toBuffer(raw)` https://github.com/ethereumjs/ethereumjs-util/blob/0bc0616c2de0de0723f87c3b64923bb9f962a8e1/index.js#L153-L175

Reply:
nice!